lib/client/doc.js

```diff
@@ -528,7 +528,7 @@ function transformX(client, server) {
   }
 
   // Ignore no-op coming from server
-  if (!server.op) return;
+  if (!('op' in server)) return;
```

Comment:
As I mentioned in the issue, can we please update `setNoOp` to return a no-op consistent with this new shape (ie `{op: null}`), and then keep this check as it was before.

Reply:
Sorry, I'm a bit confused by this. Right now `setNoOp` `delete` the `op` key from the `op`. You want it to be updated to?

```javascript
function setNoOp(op) {
  op.op = null
  delete op.create;
  delete op.del;
}
```

Reply:
Yes, please. Just so that a no-op looks the same everywhere.

Reply:
Oh okay. I try this change (and putting back `if (!server.op)`) and I've got some failing specs. I think it comes from the fact that I need to update this check in `_otApply` : 

https://github.com/antoinelyset/sharedb/blob/28f1faa7818d5af0ce9f610262a180f6651f6f61/lib/client/doc.js#L570

If I don't, `length` will fail here: 

https://github.com/sharedb/sharedb/blob/28f1faa7818d5af0ce9f610262a180f6651f6f61/lib/client/doc.js#L594

Right? But now we're back to the problem of `null` being skipped?

I'm a bit lost by this `setNoOp`, `null` is only a NoOp in JSON1 and ShareDB should not know this and continue as if it was a regular operation.

Reply:
That second link is broken but, I assume you meant to point [here](https://github.com/antoinelyset/sharedb/blob/28f1faa7818d5af0ce9f610262a180f6651f6f61/lib/client/doc.js#L594)? `length` shouldn't fail, because of the `defaultType` check (where the `defaultType` is always `json0`).

Reply:
Yes sorry, it points here. the `op` can be a noOp and still be in `json0` no? This is what I have in the specs.

Reply:
I've pushed my branch, if you want to take a look. I think I didn't quite understand.

Reply:
I think it's fine to update the check and just skip that block if `op.op === null`. All the block is doing is fragmenting the `json0` op and applying it one component at a time. Given that a no-op has no components, it's fine to just skip, and the no-op will be "applied" [below](https://github.com/antoinelyset/sharedb/blob/28f1faa7818d5af0ce9f610262a180f6651f6f61/lib/client/doc.js#L621).

Reply:
Hmm I've update `_otApply` check for a `op.op === null` but now I've got a lot of other tests failing. I'm missing something again I think.